app/models/measurement_unit.rb

```diff
@@ -15,12 +15,24 @@ class MeasurementUnit < Sequel::Model
   delegate :description, to: :measurement_unit_description
 
   class << self
-    def measurement_unit(unit_code, unit_key)
-      measurement_units[unit_key] ||= build_missing_measurement_unit(unit_code, unit_key)
+    def units(unit_code, unit_key)
+      unit = measurement_units[unit_key] ||= build_missing_measurement_unit(unit_code, unit_key)
```

Comment:
it would be much clear if we do not memoize something and returns something else.

One possibility is to leave the `measurement_unit` method as it is and create another method  that returns 
``` 
if unit['compound_units'].present?
        compound_units_for(unit)
      else
        [unit]
      end
```
or maybe you can just remove the memoization and split this line
`unit = measurement_units[unit_key] ||= build_missing_measurement_unit(unit_code, unit_key)` 
which is a bit confusing (you have to think about the priority of the operator `=` and `||=` )

Reply:
I suspect the memoization was done to reduce the database call/construction of the fallback when there is a missing measure unit in the file or the database

Reply:
I've removed this now